rust/arrow/src/array/array.rs

```diff
@@ -1267,6 +1267,36 @@ impl<OffsetSize: OffsetSizeTrait> GenericBinaryArray<OffsetSize> {
         GenericBinaryArray::<OffsetSize>::from(array_data)
     }
 
+    fn from_opt_vec(v: Vec<Option<&[u8]>>, data_type: DataType) -> Self {
```

Comment:
is this code tested somewhere?

Reply:
It is tested (indirectly) in https://github.com/apache/arrow/pull/8303/files#diff-d7b0b7cde1850e8744ceda458c6dea81R700 -- but I think a more specific test would be valuable. I will add one.

Reply:
This turned out to be a great call @jorgecarleitao  -- I found a bug in this implementation while writing a test. Thank you for the suggestion. 💯